lib/internal/modules/cjs/loader.js

```diff
@@ -336,7 +336,7 @@ function findLongestRegisteredExtension(filename) {
 // This only applies to requests of a specific form:
 // 1. name/.*
 // 2. @scope/name/.*
-const EXPORTS_PATTERN = /^((?:@[^./@\\][^/@\\]*\/)?[^@./\\][^/\\]*)(\/.*)$/;
+const EXPORTS_PATTERN = /^((?:@[^@/\\%][^@/\\%]*\/)?[^@/\\%]+)(\/.*)$/;
```

Comment:
Hm. Are we sure we want this pattern? Allowing leading dots leads to weird things like this:

```
> '../'.match(/^(?<name>(?<scope>@[^@/\\%][^@/\\%]*\/)?(?<lname>[^@/\\%]+))(?<expansion>\/.*)$/).groups
[Object: null prototype] {
  scope: undefined,
  lname: '..',
  name: '..',
  expansion: '/'
}
```

E.g. `..` is now a valid "package name".

```
> '@./../'.match(/^(?<name>(?<scope>@[^@/\\%][^@/\\%]*\/)?(?<lname>[^@/\\%]+))(?<expansion>\/.*)$/).groups
[Object: null prototype] {
  scope: '@./',
  lname: '..',
  name: '@./..',
  expansion: '/'
}
```

Reply:
```suggestion
const EXPORTS_PATTERN = /^((?:@[^@/\\%][^@/\\%]*\/)?[^@/\\%.][^@/\\%]*)(\/.*)$/;
```

We could disallow `.` as a leading character for the non-scope name only maybe?

Reply:
Thanks, agreed! Will add that now.